src/main/java/com/mihir/pipposample/Employee/EmployeeDao.java

```diff
@@ -0,0 +1,180 @@
+            throw new Exception(e);
+        }
+        return employee;
+    }
+
+    public Set<Document> bulkGetEmployeeByEmail(Set<String> emails) throws Exception{
+        if(status == HEALTH_STATUS.FAILED || mongo == null)
+            throw new Exception("Unable to connect to MongoDao");
+
+        Set<Document> employees = new HashSet<>();
+        List<Document> emailsToFind = new ArrayList<>();
+        emails.forEach(e -> emailsToFind.add(new Document("email",e)));
+
+        try{
+            MongoDatabase db = mongo.getDatabase(database);
+            MongoCollection<Document> collection = db.getCollection(empCollection);
+            Iterator iter = collection.find(new Document("$or", emailsToFind)).iterator();
+            while(iter.hasNext()) {
+                employees.add((Document) iter.next());
+            }
+        } catch (MongoException e) {
+            throw new Exception(e);
+        }
+        return employees;
+    }
+
+    public void updateEmployeeByEmail(String email, Document updateData) throws Exception{
+        if(status == HEALTH_STATUS.FAILED || mongo == null)
+            throw new Exception("Unable to connect to MongoDao");
+
+        try{
+            MongoDatabase db = mongo.getDatabase(database);
+            MongoCollection<Document> collection = db.getCollection(empCollection);
+            collection.updateOne(new Document("email", email), new Document("$set", updateData));
+        } catch (MongoException e) {
+            throw new Exception(e);
+        }
+    }
+
+    public void deleteEmployeeByEmail(String email) throws Exception{
+        if(status == HEALTH_STATUS.FAILED || mongo == null)
+            throw new Exception("Unable to connect to MongoDao");
+
+        try{
+            MongoDatabase db = mongo.getDatabase(database);
+            MongoCollection<Document> collection = db.getCollection(empCollection);
+            collection.deleteMany(new Document("email", email));
+        } catch (MongoException e) {
+            throw new Exception(e);
+        }
+    }
+
+    public void bulkDeleteEmployeeByEmail(Set<String> emails) throws Exception{
+        if(status == HEALTH_STATUS.FAILED || mongo == null)
+            throw new Exception("Unable to connect to MongoDao");
+
+        List<Document> emailsToDelete = new ArrayList<>();
+        emails.forEach(e -> emailsToDelete.add(new Document("email",e)));
+        try{
+            MongoDatabase db = mongo.getDatabase(database);
+            MongoCollection<Document> collection = db.getCollection(empCollection);
+            collection.deleteMany(new Document("$or", emailsToDelete));
+        } catch (MongoException e) {
+            throw new Exception(e);
+        }
+    }
+
+    // Check if mongo connection lives
+    @Override
+    public void serverHearbeatStarted(
+            ServerHeartbeatStartedEvent serverHeartbeatStartedEvent
+    ) {
+        synchronized (healthCheckLock){ this.status = HEALTH_STATUS.STARTED; }
+    }
```

Comment:
when you write a mongoclientprovider, this will become. the injector's responsibility
https://github.com/unbxd/banshee/blob/1d94ccaed724a8618eb81c057d35eb02a7143698/commons/src/main/java/com/unbxd/commons/guice/providers/MongoClientProvider.java#L23-L22